lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/src/main/java/io/lighty/applications/rnc/module/RncLightyModule.java

```diff
@@ -94,56 +96,55 @@ public boolean initModules() {
         return true;
     }
 
-    private LightyController initController(ControllerConfiguration config) throws RncLightyAppStartException {
-        LightyControllerBuilder lightyControllerBuilder = new LightyControllerBuilder();
+    private LightyController initController(final ControllerConfiguration config) throws RncLightyAppStartException {
         try {
-            return lightyControllerBuilder.from(config).build();
+            return new LightyControllerBuilder().from(config).build();
         } catch (ConfigurationException e) {
             throw new RncLightyAppStartException("Unable to initialize lighty.io controller module!", e);
         }
     }
 
-    private NetconfSBPlugin initNetconf(NetconfConfiguration config, LightyServices services)
+    private NetconfSBPlugin initNetconf(final NetconfConfiguration config, final LightyServices services)
             throws RncLightyAppStartException {
         try {
-            NetconfConfiguration configWithServices =
+            final NetconfConfiguration configWithServices =
                     NetconfConfigUtils.injectServicesToTopologyConfig(config, services);
             return NetconfTopologyPluginBuilder.from(configWithServices, services).build();
         } catch (ConfigurationException e) {
             throw new RncLightyAppStartException("Unable to initialize lighty.io NETCONF module!", e);
         }
     }
 
-    private CommunityRestConf initRestconf(RncRestConfConfiguration config, LightyServices services) {
-        final RncRestConfConfiguration confConf = RncRestConfConfigUtils.getRestConfConfiguration(config, services);
+    private CommunityRestConf initRestconf(final RestConfConfiguration config, final LightyServerConfig serverConfig,
+            final LightyServices services) {
+        final RestConfConfiguration restConfConfiguration = RestConfConfigUtils.getRestConfConfiguration(config,
+                services);
         final InetSocketAddress inetSocketAddress =
-                new InetSocketAddress(confConf.getInetAddress(), confConf.getHttpPort());
+                new InetSocketAddress(config.getInetAddress(), config.getHttpPort());
 
-        if (confConf.isUseHttps()) {
-            this.jettyServerBuilder = new HttpsLightyServerBuilder(inetSocketAddress, config.getSecurityConfig());
+        if (serverConfig.isUseHttps()) {
+            this.jettyServerBuilder = new HttpsLightyServerBuilder(inetSocketAddress, serverConfig.getSecurityConfig());
         } else {
             this.jettyServerBuilder = new LightyServerBuilder(inetSocketAddress);
         }
 
-        return CommunityRestConfBuilder.from(confConf)
+        return CommunityRestConfBuilder.from(restConfConfiguration)
```

Comment:
What about?

```
    private CommunityRestConf initRestconf(final RestConfConfiguration config, final LightyServerConfig serverConfig) {
        final InetSocketAddress inetSocketAddress =
                new InetSocketAddress(config.getInetAddress(), config.getHttpPort());

        if (serverConfig.isUseHttps()) {
            this.jettyServerBuilder = new HttpsLightyServerBuilder(inetSocketAddress, serverConfig.getSecurityConfig());
        } else {
            this.jettyServerBuilder = new LightyServerBuilder(inetSocketAddress);
        }

        return CommunityRestConfBuilder.from(config)
            .withLightyServer(jettyServerBuilder)
            .build();
    }
```

Reply:
If you mean by this to remove from code:
```java
final RestConfConfiguration restConfConfiguration = RestConfConfigUtils.getRestConfConfiguration(config, services);
```

Then this is how lighty services are copied to Restconf configuration.
https://github.com/PANTHEONtech/lighty/blob/master/lighty-modules/lighty-restconf-nb-community/src/main/java/io/lighty/modules/northbound/restconf/community/impl/util/RestConfConfigUtils.java#L134

https://github.com/PANTHEONtech/lighty/blob/48c87f70707d06d03dc2f1c16af7468902bd1012/lighty-examples/lighty-community-restconf-actions-app/src/main/java/io/lighty/examples/controllers/actions/Main.java#L154

Reply:
OK